### ISSUE TITLE
prevent getting zero p and q

### ISSUE THREAD

**Issue by ramdhan1989** [alkaline-ml/pmdarima#386]
I think searching hyperparam process always starts from zero. How to prevent of getting zero of p and q values ? setting values above zero for start_p and start_q didn't help in this case. 

thank you

please advice

**Comment by aaronreidsmith** [comment]
Hey @ramdhan1989 could you provide some reproducible code and the output of `python -m "import pmdarima; pmdarima.show_versions()`? It will help us a lot in answering your question

**Comment by ramdhan1989** [comment]
this is the print out : 
System:
   machine: Windows-10-10.0.18362-SP0
    python: 3.5.6 |Anaconda, Inc.| (default, Aug 26 2018, 16:05:27) [MSC v.1900 64 bit (AMD64)]

Python dependencies:
     pandas: 0.25.3
      numpy: 1.18.4
   pmdarima: 1.7.0
      scipy: 1.4.1
        pip: 10.0.1
statsmodels: 0.11.1
     joblib: 0.14.1
 setuptools: 40.2.0
    sklearn: 0.22.2.post1
     Cython: 0.29.17

thank you

**Comment by tgsmith61591** [comment]
> How to prevent of getting zero of p and q values?

Short answer is you cannot guarantee this. There are a couple reasons for this:

* The initial stepwise model `auto_arima` fits is a null model `(0,d,0)(0,D,0,m)` for the sake of simplicity. That's the baseline model all others must beat. There is no way to guarantee this will not be the best model
* We only constrain max `p`, `q`, `P` and `Q`. This is because it is (nearly) always preferable to favor simpler models
* This is the R package's behavior, and we try to adhere as closely as possibly to it.

Prior to the `1.5.2` version, we started random searches at `start_p`, `start_q`, etc. However, after that we switched to starting at zero, since that's what the R package's search does. In fact, I've considered deprecating the `start_p`, `start_q`, `start_P` and `start_Q` params since there are so few cases where they're actually used.

**Comment by tgsmith61591** [comment]
As a quick addendum, this is where we use the start params in the stepwise:
https://github.com/alkaline-ml/pmdarima/blob/master/pmdarima/arima/auto.py#L650

They seed `p,q,P,Q` and then are no longer used after that.

```python
        # adjust to p, q, P, Q vals
        p = start_p = min(start_p, max_p)
        q = start_q = min(start_q, max_q)
        P = start_P = min(start_P, max_P)
        Q = start_Q = min(start_Q, max_Q)
```

**Comment by ramdhan1989** [comment]
thanks for your explanation

**Comment by j-adamczyk** [comment]
@tgsmith61591 I got to this issue since I thought this was a bug. As far I as know, this is not mentioned anywhere in the docs and can lead to unexpected behavior (as it did for my case). This should be explicitly stated in the docs.
